src/subapps/studioLegacy/containers/StudioListContainer.tsx

```diff
@@ -47,128 +69,46 @@ const StudioListContainer: React.FunctionComponent<{
-        )
-      );
-      setResources({
-        next: response._next || null,
-        resources: [
-          ...resources,
-          ...(newResources as Resource<{
-            label: string;
-            description: string;
-          }>[]),
-        ],
-        total: response._total,
-        busy: false,
-        error: null,
-      });
-    } catch (error) {
-      setResources({
-        next,
-        error,
-        resources,
-        total,
-        busy: false,
-      });
+    if (data?._next) {
+      setNext(parseInt(data?._next, 10));
     }
   };
 
-  React.useEffect(() => {
-    setResources({
-      next,
-      resources,
-      total,
-      busy: true,
-      error: null,
-    });
-
-    let response: PaginatedList<Resource>;
-
-    // get all resources of type studio
-    nexus.Resource.list(orgLabel, projectLabel, {
-      type: DEFAULT_STUDIO_TYPE,
-      deprecated: false,
-      size: 10,
-      q: searchQuery,
-    })
-      .then(studioResponse => {
-        response = studioResponse;
-        return Promise.all(
-          response._results.map(resource =>
-            nexus.Resource.get(
-              orgLabel,
-              projectLabel,
-              encodeURIComponent(resource['@id'])
-            )
-          )
-        );
-      })
-      .then(studios => {
-        setResources({
-          next: response._next || null,
-          resources: studios as Resource<{
-            label: string;
-            description?: string;
-          }>[],
-          total: response._total,
-          busy: false,
-          error: null,
-        });
-      })
-      .catch(error => {
-        setResources({
-          next,
-          error,
-          resources,
-          total,
-          busy: false,
-        });
-      });
-  }, [orgLabel, projectLabel, searchQuery]);
-
   return (
-    <StudioList
-      studios={resources.map(r => ({
-        id: r['@id'],
-        name: r.label,
-        description: r.description,
-      }))}
-      onLoadMore={handleLoadMore}
-      searchQuery={searchQuery}
-      makeResourceUri={makeStudioUri}
-      total={total}
-      busy={busy}
-      error={error}
-      goToStudio={(id: string) => goToStudio(id)}
-      createStudioButton={
-        <CreateStudioContainer
-          orgLabel={orgLabel}
-          projectLabel={projectLabel}
-          goToStudio={goToStudio}
-        />
-      }
-    ></StudioList>
+    <div className="studio-list" id={'studio-list'}>
+      <CreateStudioContainer
+        orgLabel={orgLabel}
+        projectLabel={projectLabel}
+        goToStudio={goToStudio}
+      />
+      <br />
+      <InfiniteSearch
+        dataLength={data?._total || 0}
+        onLoadMore={handleLoadMore}
+        hasMore={(data?._results.length || 0) < Number(data?._total || 0)}
+        defaultSearchValue={searchQuery}
+        height={500}
+      >
+        {status === 'error' || listError !== null ? (
+          <Empty description="Sorry, something went wrong" />
```

Comment:
remove this

Reply:
We will have to keep the first part because we want to display           <Empty description="Sorry, something went wrong" /> based on the status. I am removing the second part.